tests/basic_sanity_test.go

```diff
@@ -16,6 +16,7 @@ const (
 
 var _ = Describe(TestSuiteName, func() {
 	f := framework.NewFramework("sanity")
+	f.SkipNamespaceCreation = true
```

Comment:
I see k8s does it this way, meaning define an exported field in Framework and have the caller of `NewFramework` set this field after Framework is created. What do you think about passing in this bool to `NewFramework`?

Reply:
So coming from a java background, there I would make an overloaded constructor with multiple parameters. But in GOLANG that is not an option. So we would have to create a NewFrameworkWithOptions or something similar. Then if we add another field we will need another NewFrameworkSomethingOrOther, and it can get messy really quick. So I understand why the k8s people went with exposing the flag and setting it after creating the new instance.

So personally I am fine with either NewFrameworkWithOptions, and stuffing all the flags in the options, and then having the NewFrameworkWithOptions copy over the flags into the real struct. Or just exposing the flags and having the caller set them after NewFramework.

If we want to discuss on picking one or the other we can do that, but not sure if it is that important.

Reply:
I'm ok with the way it is, it seems cleaner to do this way

Reply:
@awels I like your suggestion and just read similar code in the golang xz compression pkg. See, for instance, [NewWriter](https://github.com/ulikunitz/xz/blob/master/lzma/writer.go#L158) and how it calls an exported constructor method on a config struct. Essentially a default constructor can be called as a simple func, or you can fill-in a public config struct and call new on that. I think this is quite clean.

Reply:
Done